tests/src/test/java/alluxio/master/block/MultiWorkerIntegrationTest.java

```diff
@@ -61,7 +60,9 @@ public void writeLargeFile() throws Exception {
             .setLocationPolicy(new RoundRobinPolicy()));
     URIStatus status = fs.getStatus(file);
     assertEquals(100, status.getInAlluxioPercentage());
-    try (FileInStream inStream = fs.openFile(file)) {
+    // Use NO_CACHE to prevent evicting "none-local" blocks
+    try (FileInStream inStream = fs
```

Comment:
@calvinjia , it seems if multiple workers are on the same node as the client, only one is considered as local and other workers are considered "remote" and triggering async cache. I think we should fix this (perhaps separately). what do you think?

Reply:
Do you mean the behavior should be all workers are considered local, what would be involved in the fix? I don't think we have changed how we calculate locality?

Reply:
@calvinjia I think I found a better fix. Can you double check commit 201c612 and verify this is a legit fix?
Essentially, we should not async-cache if the BlockInStream is already local